tests/suite_test.go

```diff
@@ -176,23 +195,83 @@ func (suite *IntegrationTestSuite) SetupSuite() {
 	require.NoError(suite.T(), err, fmt.Sprintf("Error Deploying DSPA:\n%s", testUtil.PrintConditions(ctx, DSPA, DSPANamespace, clientmgr.k8sClient)))
 	loggr.Info("DSPA Deployed.")
 
-	loggr.Info("Setting up Portforwarding service.")
-	options := []*forwarder.Option{
-		{
-			LocalPort:   PortforwardLocalPort,
-			RemotePort:  APIServerPort,
-			ServiceName: fmt.Sprintf("ds-pipeline-%s", DSPA.Name),
-			Namespace:   DSPANamespace,
-		},
+	if endpointType == "service" {
+
+		loggr.Info("Setting up Portforwarding service.")
+		options := []*forwarder.Option{
+			{
+				LocalPort:   PortforwardLocalPort,
+				RemotePort:  APIServerPort,
+				ServiceName: fmt.Sprintf("ds-pipeline-%s", DSPA.Name),
+				Namespace:   DSPANamespace,
+			},
+		}
+
+		forwarderResult, err = forwarder.WithForwarders(ctx, options, kubeconfig)
+		suite.Require().NoError(err)
+		_, err = forwarderResult.Ready()
+		suite.Require().NoError(err)
+
+		APIServerURL = fmt.Sprintf("http://127.0.0.1:%d", PortforwardLocalPort)
+
+		loggr.Info(fmt.Sprintf("Port forwarding service Successfully set up: %s", APIServerURL))
+
+	} else if endpointType == "route" {
+
+		// Define the namespaced name
+		key := types.NamespacedName{
+			Namespace: DSPANamespace,
+			Name:      "ds-pipeline-" + suite.DSPA.Name,
+		}
+
+		route := &routev1.Route{}
+
+		// Retrieve the router
+		count := 1
+		ok := false
+		for count < 10 {
+			err := clientmgr.k8sClient.Get(context.TODO(), key, route)
+			if err == nil {
+				ok = true
+				break
+			}
+			count++
+			time.Sleep(1 * time.Second)
+		}
+		if !ok {
+			log.Fatal("failed to retrieve router")
+		}
```

Comment:
dspa has already readied here: https://github.com/opendatahub-io/data-science-pipelines-operator/pull/711/files#diff-96e66e1ccb0083ac5a2b7fc37cb002803d242ea67c1981a5d09ef8ee2668354cR194

so the route should already have been created, so we shouldn't need to have retry attempts here, it should just work the first time.

Reply:
No. It is not like that. It is async and the route can take time to be up and running. I implemented the code after seeing failures in the test suite.